zk/conn.go

```diff
@@ -317,6 +326,65 @@ func (c *Conn) connect() error {
 	}
 }
 
+func (c *Conn) resendZkAuth(reauthReadyChan chan struct{}) {
+
+	c.credsMu.Lock()
+	if len(c.creds) > 0 {
+		c.logger.Printf("Re-submitting `%d` credentials after reconnect",
+			len(c.creds))
+		for _, cred := range c.creds {
+			resChan, err := c.sendRequest(
+				opSetAuth,
+				&setAuthRequest{Type: 0,
+					Scheme: cred.scheme,
+					Auth:   cred.auth,
+				},
+				&setAuthResponse{},
+				nil)
+
+			if err != nil {
+				c.logger.Printf("Call to sendRequest failed during credential resubmit: %s", err)
+				// FIXME(prozlach): lets ignore errors for now
+				continue
+			}
+
+			res := <-resChan
+			if res.err != nil {
+				c.logger.Printf("Credential re-submit failed: %s", res.err)
+				// FIXME(prozlach): lets ignore errors for now
+				continue
+			}
+		}
+	}
+	c.credsMu.Unlock()
+	close(reauthReadyChan)
```

Comment:
if you defer this close too, you can do:

if len(c.creds) == 0 {
   return
}

and avoid a level of indentation